File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/TweetIdentifier.cs
namespace WorldFeed.Common.Public.Models
{
    using System.Globalization;

    using WorldFeed.Common.Public.Models.Interfaces;

    public class TweetIdentifier : ITweetIdentifier
    {
        public TweetIdentifier(long tweetId)
        {
            Id = tweetId;
            IdStr = tweetId.ToString(CultureInfo.InvariantCulture);
        }

        public long Id { get; set; }
        public string IdStr { get; set; }

        public override string ToString()
        {
            return IdStr ?? Id.ToString();
        }
    }
}

File: Sharebook/Client/src/app/shared/models/TweetIdentifier.ts
import {ITweetIdentifier} from "../posts/models/tweet-identifier";

export class TweetIdentifier implements ITweetIdentifier {
  constructor(tweetId: number) {
    this.id = tweetId;
    this.idStr = tweetId.toString(/*CultureInfo.InvariantCulture*/);
  }

  public id: number;
  public idStr: string;

  public toString(): string {
    return this.idStr ?? this.id.toString();
  }
}
